modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java

```diff
@@ -542,8 +544,7 @@ private CompletableFuture<Object> processMultiEntryAction(ReadWriteMultiRowRepli
                     upsertLockFuts[i++] = takeLocksForUpsert(searchKey, indexId, txId);
                 }
 
-                return CompletableFuture.allOf(upsertLockFuts).thenApply(ignore -> {
-                    Collection<BinaryRow> rowsToInsert = new ArrayList<>();
+                return CompletableFuture.allOf(upsertLockFuts).thenCompose(ignore -> {
                     Map<RowId, BinaryRow> rowsToUpdate = new HashMap<>();
 
                     int futNum = 0;
```

Comment:
Please avoid joins on lockFutures arrays, e.g. by using CompletableFutures.allOf(). We should have consistent behavior  and proper exception handling both within single and multiple operations processing, meaning that we should prefer chaining of the futures like it's done for single oprations.

Reply:
Here I nothing changed except return value.
I didn't untested what you mean, but it is not related to this pr.

Reply:
Never mind. My fault.